sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -283,7 +283,19 @@ private[sql] abstract class BaseWriterContainer(
   }
 
   private def newOutputCommitter(context: TaskAttemptContext): FileOutputCommitter = {
```

Comment:
I used `FileOutputCommitter` here because we need to retrieve the actual path of file being written, which is returned by `FileOutputCommitter.getWorkPath`. This implies customized output committers must be subclasses of `FileOutputCommitter`, which was true for `DirectParquetOutputCommitter`. But this restriction seems too strict. Resorting to `OutputCommitter` rather than `FileOutputCommitter` in another PR.